packages/kbn-io-ts-utils/src/strict_keys_rt/index.ts

```diff
@@ -62,7 +62,7 @@ function getHandlingTypes(type: t.Mixed, key: string, value: object): t.Mixed[]
       return getHandlingTypes(type.type, key, value);
 
     case 'UnionType':
-      const matched = type.types.find((m) => m.is(value));
+      const matched = type.types.find((m) => isRight(m.decode(value)));
```

Comment:
`m.is(value)` was only working for primitive types.

Reply:
I don't think that should be the case. The intent of `is` is to be type guard. Where are you seeing this fail, with custom runtime types or also with built-ins like `rt.type({})`?

Reply:
There is a [test](https://github.com/elastic/kibana/pull/143190/files#diff-1a9968dc7eaac2daf3d7cf937710c62173d809f3cf09701e768cafb73becc035R17-R32) for the scenario I've encountered. 
It fails with custom types within an union. 

From what I understand, the input provided to the `getHandledKeys` is the raw input (not validated, i.e. decoded). Therefore, when iterating over the union types and trying to match the type with the type guard, it cannot work for custom types as the value is still unknown at this time. 
When using built-ins `t.type()` it works fine.

Reply:
Yeah, I think you're right. The issue is with types that convert, not simply validate (the latter is what most built-in types do). Thanks for catching this! Do you mind replacing the usage of a custom runtime type in the test with `isoToEpochRt`? It should have the same issue and would remove the need to implement a custom runtime type in the test.

Reply:
I'll make the change for `isoToEpochRt`